### PR TITLE
feat(externals): apply `production` condition to package.exports

### LINKED ISSUE #844: apply `production` contraint to traced `package.json` files
We trace and extract package files with `production` condition however node, does not apply it by default without passing `-C production` flag and we cannot always pass a node.js flag.

When extracting packages, we shall apply it to `exports` that production is picked by default.

More context: https://github.com/unjs/nitro/pull/605#issuecomment-1286660793

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

resolves #844

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Rewrite `package.json` with production condition in exports field before writing to be consistent with the condition of resoluton before. Tested against two linked packages.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/unit/externals.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      name: "vue-router@4.1.6",
+      in: {
+        ".": {
+          types: "./dist/vue-router.d.ts",
+          node: {
+            import: {
+              production: "./dist/vue-router.node.mjs",
+              development: "./dist/vue-router.node.mjs",
+              default: "./dist/vue-router.node.mjs",
```
Comment (danielroe) [inline test/unit/externals.test.ts:15]:
```suggestion
              production: "./dist/vue-router.node.mjs",
              development: "./dist/vue-router.dev.node.mjs",
              default: "./dist/vue-router.dev.node.mjs",
```

**Review by danielroe — APPROVED** [review]
Nice. 👌

File: test/unit/externals.test.ts (hunk 1 above)
Comment (pi0) [reply test/unit/externals.test.ts:15]: I took it from here https://unpkg.com/browse/vue-router@4.1.6/package.json it doesn't have dev however would be nice if it had. Are you suggesting this for test clarity?

Comment (danielroe) [reply test/unit/externals.test.ts:15]: Exactly. Otherwise it doesn't show a difference between in and out.

**Pull request merged**